lib/profile/node.rb

```diff
@@ -122,15 +125,16 @@ def status
     end
 
     attr_reader :name
-    attr_accessor :hostname, :identity, :deployment_pid, :exit_status, :hunter_label
+    attr_accessor :hostname, :identity, :deployment_pid, :exit_status, :hunter_label, :ip
 
-    def initialize(hostname:, identity: nil, deployment_pid: nil, exit_status: nil, hunter_label: nil, name: nil)
+    def initialize(hostname:, identity: nil, deployment_pid: nil, exit_status: nil, hunter_label: nil, name: nil, ip: "Unknown")
```

Comment:
Why do we want `ip` to be `'Unknown'` if it isn't passed? How does this affect the IP being saved to the Ansible inventory?

Reply:
I considered it good practice as a default value. A node is required to have at least some value for `ip` so that the keys in the `groups` hash can still be used with the new format. The inventory file would also read
```
matt-test1.novalocal ansible_host=Unknown
```
which I imagine would cause issues for Ansible, but if a node is lacking an IP then that's inevitably going to cause issues somewhere and this seems like a decent place to diagnose an issue.

Reply:
This is something that could become rather complicated, due to the way that `use_hunter` is implemented (I'd like to improve it sometime in the near future). For now, please could we assume that:
-  if a user has enabled `use_hunter`, it is assumed that they will always have it enabled, as the codebase is split in a way that makes it hard to switch between the two modes

and only save/use the node's IP address if `use_hunter` is enabled? I realise this creates a touch of tech debt if/when we get around to evaluating how we can improve the Hunter integration, but it solves the problem for now.